### PR TITLE
groove: init at 5.7.4

### PR DESCRIPTION
**Pull request by jfrankenau**
###### Motivation for this change

A graph transformation tool useful for model checking.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/science/programming/groove/default.nix [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+    mkdir -p $out/share/applications
+    ln -s ${desktopItem}/share/applications/* $out/share/applications/
+
+    mkdir -p $out/share/icons/hicolor/{16x16,32x32}/apps
+    icotool -x -i 1 -o $out/share/icons/hicolor/32x32/apps/groove.png groove-green-g.ico
+    icotool -x -i 2 -o $out/share/icons/hicolor/16x16/apps/groove.png groove-green-g.ico
+  '';
+
+  meta = with stdenv.lib; {
```
Comment (aanderse) [inline pkgs/applications/science/programming/groove/default.nix]: Please add `platforms`

File: pkgs/applications/science/programming/groove/default.nix [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    name = "groove-simulator";
+    exec = "groove-simulator";
+    icon = "groove";
+    desktopName = "GROOVE Simulator";
+    comment = "GRaphs for Object-Oriented VErification";
+    categories = "Science;ComputerScience;";
+  };
+
+in stdenv.mkDerivation rec {
+  name = "groove-${version}";
```
Comment (aanderse) [inline pkgs/applications/science/programming/groove/default.nix]: Please change to `pname = "groove";`

**Review by aanderse — CHANGES_REQUESTED** [review]

**Comment by jfrankenau** [comment]
@aanderse, thanks for the review. I've included your changes.

File: pkgs/applications/science/programming/groove/default.nix [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    comment = "GRaphs for Object-Oriented VErification";
+    categories = "Science;ComputerScience;";
+  };
+
+in stdenv.mkDerivation rec {
+  pname = "groove";
+  version = "5.7.4";
+
+  src = fetchurl {
+    url = "mirror://sourceforge/groove/groove/${version}/${builtins.replaceStrings ["."] ["_"] name}-bin.zip";
```
Comment (aanderse) [inline pkgs/applications/science/programming/groove/default.nix]: Oops! Looks like now we want something like `url = "mirror://sourceforge/groove/groove/${version}/${pname}-${builtins.replaceStrings ["."] ["_"] version}-bin.zip";` here instead.

**Review by aanderse — CHANGES_REQUESTED** [review]

File: pkgs/applications/science/programming/groove/default.nix (hunk 3 above)
Comment (jfrankenau) [reply pkgs/applications/science/programming/groove/default.nix]: Ups, sorry. I had an overlay active which shadowed my local nixpkgs checkout.

**Review by aanderse — APPROVED** [review]

**Comment by aanderse** [comment]
Thanks for your contribution!

**Pull request merged**